documentation/manual/src/ornate/overview.md

```diff
@@ -47,25 +37,80 @@ And also:
+it to the server, and eventually decodes the HTTP response (according to the endpoint description).
 
-The following code, located in the `server` sub-project, defines the implementation of
-the web service.
+## Server implementation
+
+Similarly, a server implementation of the endpoints can be obtained by mixing the appropriate
+interpreters to the `CounterEndpoints` trait. In this example, we get a JVM server that uses
+Play framework under the hood:
 
 ~~~ scala src=../../../examples/overview/server/src/main/scala/overview/CounterServer.scala#relevant-code
 ~~~
 
-The `CounterServer.routes` value is just a `play.api.routing.Router.Routes`.
-To get an executable Web server we need to setup a “main” like the following:
+The `CounterServer.routes` value produced by the *endpoints* library is a `Routes` value directly
+usable by Play framework. To get an executable Web server we can setup a “main” like the following:
 
 ~~~ scala src=../../../examples/overview/server/src/main/scala/overview/Main.scala#relevant-code
 ~~~
 
-## What else?
+The routes implementations provided by *endpoints* decode the incoming HTTP requests, call the corresponding logic
+(here, incrementing the counter or getting its current value), and build the HTTP responses.
+
+## Documentation generation
+
+We conclude this overview by showing how to generate documentation for the endpoints, again by mixing the appropriate
+interpreters:
+
+~~~ scala src=../../../examples/overview/documentation/src/main/scala/overview/CounterDocumentation.scala#relevant-code
+~~~
+
+This code defines a `CounterDocumentation` object with an `api` member containing an OpenAPI object documenting
+the `currentValue` and `increment` endpoints.
 
-You can also get a Scala/JVM client (which uses `play-ws` under the hood) as follows:
+We can then render the documentation, for instance in JSON:
 
-~~~ scala src=../../../examples/overview/play-client/src/main/scala/overview/CounterClient.scala#relevant-code
+~~~ scala src=../../../examples/overview/documentation/src/main/scala/overview/CounterDocumentation.scala#export
 ~~~
 
-Thus, you can distribute a (fully working) JVM client, which is independent of your implementation.
+This prints the following OpenAPI document:
+
+~~~ javascript
+{
+  "openapi" : "3.0.0",
+  "info" : {
+    "title" : "API to manipulate a counter",
+    "version" : "1.0.0"
+  },
+  "paths" : {
+    "/increment" : {
+      "post" : {
+        "responses" : {
+          "200" : {
+            "description" : ""
+          }
+        },
+        "requestBody" : {
+          "content" : {
+            "application/json" : {}
+          }
+        }
+      }
+    },
+    "/current-value" : {
+      "get" : {
+        "responses" : {
+          "200" : {
+            "description" : "",
+            "content" : {
+              "application/json" : {}
+            }
+          }
+        }
+      }
+    }
+  }
+}
+~~~ 
```

Comment:
I wish I could generate this JSON document from our code, but that would require a bit more work. So, I’ve manually copy-pasted the JSON document.